kubernetes/schema_pod_template.go

```diff
@@ -0,0 +1,21 @@
+package kubernetes
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func podTemplateFields(isUpdatable bool) map[string]*schema.Schema {
+	s := map[string]*schema.Schema{
+		"metadata": metadataSchema("stateful set", true),
+		"spec": {
+			Type:        schema.TypeList,
+			Description: "Spec of the pods owned by the stateful set",
+			Optional:    true,
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: podSpecFields(false),
```

Comment:
`isUpdatable` should have been used here.

Reply:
If I remember correctly, the K8S is restricting updates on StatefulSet to just a few attributes and the `spec` is not one of them.
Let me double check which ones exactly.

Reply:
The only fields that can be updated on a StatefulSet spec without recreation of the StatefulSet are:
- `replicas`
- `template`
- `updateStrategy`

cf. https://github.com/kubernetes/kubernetes/blob/v1.10.0/pkg/apis/apps/validation/validation.go#L153

So everything under `template.spec` is supposed to updatable.

The API doc is not explicit about that: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#statefulset-v1-apps

Not being able to update pod spec fields without recreation makes sense on an independent Pod, but on a Pod spec template it does not. The controller is there to handle the recreation of the pods it manages.

Note: since kubernetes 1.7, the following documentation section was dropped:

> As demonstrated in the [Scaling a StatefulSet](#scaling-a-statefulset) section,
the `replicas` field of a StatefulSet is mutable. The only other field of a 
StatefulSet that can be updated is the `spec.template.containers` field. 

cf. : https://github.com/kubernetes/website/pull/4174/files#diff-42e39658021b706bcc9478c8cc73c4a3L452